src/main/java/edu/kis/powp/jobs2d/command/ExceedingCanvasCommandVisitor.java

```diff
@@ -0,0 +1,57 @@
+package edu.kis.powp.jobs2d.command;
+
+import javax.swing.*;
+import java.awt.Dimension;
+import java.util.Iterator;
+import java.util.logging.Logger;
+
+public class ExceedingCanvasCommandVisitor implements CommandVisitor {
+
+    private JPanel drawArea;
+    private static Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+
+    public ExceedingCanvasCommandVisitor(JPanel drawArea) {
+        this.drawArea = drawArea;
+    }
+
+    private Dimension getDrawAreaSize() {
+        return drawArea.getSize();
+    }
+
+    @Override
+    public void visit(OperateToCommand operateToCommand) {
+        Dimension size = getDrawAreaSize();
+        int width = size.width;
+        int height = size.height;
```

Comment:
A canvas should be an abstraction independent of GUI elements like JPanel. It would be preferred that an application can manage canvases of all shapes (not only rectangular like A4 and B3).